static/js/index.tsx

```diff
@@ -18,15 +18,14 @@ Sentry.init({
   environment: SETTINGS.environment
 })
 
-const store = configureStore()
+const store: any = configureStore()
```

Comment:
hum, I hoped that we could avoid `any` for redux stuff, did this not work at all without it?

Reply:
I'm seeing this error message:

    ➜  ocw-studio git:(gs/site-detail-ui) ✗ dwatch npm run typecheck
    Creating ocw-studio_watch_run ... done
    
    > ocw_studio@0.0.0 typecheck /src
    > tsc --noEmit --strictNullChecks
    
    static/js/components/SiteComponent.tsx:14:68 - error TS2345: Argument of type 'DefaultRootState' is not assignable to parameter of type 'ReduxState'.
      Type 'DefaultRootState' is missing the following properties from type 'ReduxState': queries, entities
    
    14   const website = useSelector(state => websiteSelectors.getWebsite(state, name))
                                                                          ~~~~~
    
    static/js/index.tsx:27:18 - error TS2322: Type 'Store<CombinedState<{ entities: Entities; queries: QueriesState; }>, ReduxQueryAction<Entities>> & { ...; }' is not assignable to type 'Store<any, AnyAction>'.
      Types of property 'dispatch' are incompatible.
        Type 'Dispatch<ReduxQueryAction<Entities>>' is not assignable to type 'Dispatch<AnyAction>'.
          Type 'AnyAction' is not assignable to type 'ReduxQueryAction<Entities>'.
            Type 'AnyAction' is not assignable to type 'MutateAsyncAction<Entities>'.
              Property 'url' is missing in type 'AnyAction' but required in type 'QueryConfig<Entities>'.
    
    27       <Component store={store} />
                        ~~~~~
    
      node_modules/redux-query/index.d.ts:167:5
        167     url: Url;
                ~~~
        'url' is declared here.
      static/js/Router.tsx:11:3
        11   store: Store
             ~~~~~
        The expected type comes from property 'store' which is declared here on type 'IntrinsicAttributes & RootProps & { children?: ReactNode; }'
    
    static/js/pages/SitePage.tsx:17:68 - error TS2345: Argument of type 'DefaultRootState' is not assignable to parameter of type 'ReduxState'.
    
    17   const website = useSelector(state => websiteSelectors.getWebsite(state, name))
                                                                          ~~~~~
    
    
    Found 3 errors.
    
    npm ERR! code ELIFECYCLE
    npm ERR! errno 2
    npm ERR! ocw_studio@0.0.0 typecheck: `tsc --noEmit --strictNullChecks`
    npm ERR! Exit status 2
    npm ERR! 
    npm ERR! Failed at the ocw_studio@0.0.0 typecheck script.
    npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
    
    npm ERR! A complete log of this run can be found in:
    npm ERR!     /home/node/.npm/_logs/2021-02-18T19_33_27_790Z-debug.log
    ERROR: 2

Reply:
oh ok yeah, I know what the issue is - the `store: Store` type on `App.tsx` isn't sufficiently specific. I fix that here: https://github.com/mitodl/ocw-studio/pull/66/files#diff-0dab17b074318efbcc682ae6d96b4c08de1c795a1b83f549ccac4e6f49d35937 you should be able to use the same fix, and then get specific, full inferred types for the store

Reply:
needs to be `store: ReturnType<typeof configureStore>`